payments-core/src/main/java/com/stripe/android/view/CardWidgetViewModel.kt

```diff
@@ -32,22 +30,24 @@ internal class CardWidgetViewModel(
     private val paymentConfigProvider: Provider<PaymentConfiguration>,
     private val stripeRepository: StripeRepository,
     private val dispatcher: CoroutineDispatcher = Dispatchers.IO,
-    private val handle: SavedStateHandle
 ) : ViewModel() {
 
     private val _isCbcEligible = MutableStateFlow(false)
     val isCbcEligible: StateFlow<Boolean> = _isCbcEligible
-    var onBehalfOf: String? = handle[ON_BEHALF_OF]
-        set(value) {
-            field = value
-            handle[ON_BEHALF_OF] = value
-            getEligibility()
-        }
+
+    private var _onBehalfOf: String? = null
```

Comment:
should this be a mutable stateflow?

And then you could make the isCbcEligible stateflow directly dependent on this one via a `mapAsStateFlow`, rather than needing a special set function for setting OBO that then sets isCbcEligible

Reply:
We could but there's nothing that needs to observe onBehalfOf outside of the viewmodel so it seems equivalent to me

Reply:
I think it'd be useful if we added another callsite within the viewmodel too, and it also seems more idiomatic/easier to follow